src/components/PreferencesBodyKubernetes.vue

```diff
@@ -0,0 +1,157 @@
+  },
+  methods: {
+    onChange() {
+      this.$emit('change:kubernetes-version');
+    },
+    /**
+     * Get the display name of a given version.
+     * @param version The version to format.
+     */
+    versionName(version: VersionEntry) {
+      const names = (version.channels ?? []).filter(ch => !/^v?\d+/.test(ch));
+
+      if (names.length > 0) {
+        return `v${ version.version.version } (${ names.join(', ') })`;
+      }
+
+      return `v${ version.version.version }`;
+    },
+  }
+});
+</script>
+
+<template>
+  <div class="preferences-body">
+    <input-legend>
+      <template #legend>
+        <span>Kubernetes</span>
+      </template>
+      <checkbox
+        v-model="enableKubernetes"
+        label="Enable Kubernetes"
+      />
+    </input-legend>
+    <input-legend class="w-half">
+      <template #legend>
+        Kubernetes Version
+      </template>
+      <select
+        v-model="settings.kubernetes.version"
+        class="select-k8s-version"
+        :disabled="isKubernetesDisabled"
+        @change="onChange"
+      >
+        <!--
+            - On macOS Chrome / Electron can't style the <option> elements.
+            - We do the best we can by instead using <optgroup> for a recommended section.
+            -->
+        <optgroup v-if="recommendedVersions.length > 0" label="Recommended Versions">
+          <option
+            v-for="item in recommendedVersions"
+            :key="item.version.version"
+            :value="item.version.version"
+            :selected="item.version.version === defaultVersion.version.version"
+          >
+            {{ versionName(item) }}
+          </option>
+        </optgroup>
+        <optgroup v-if="nonRecommendedVersions.length > 0" label="Other Versions">
+          <option
+            v-for="item in nonRecommendedVersions"
+            :key="item.version.version"
+            :value="item.version.version"
+            :selected="item.version.version === defaultVersion.version.version"
+          >
+            v{{ item.version.version }}
+          </option>
+        </optgroup>
+      </select>
+    </input-legend>
+    <input-legend class="w-half">
+      <template #legend>
+        Kubernetes Port
+      </template>
+      <input
+        v-model="kubernetesPort"
+        type="number"
+        :disabled="isKubernetesDisabled"
+      />
+    </input-legend>
+    <input-legend>
+      <template #legend>
+        <span>Traefik</span>
+      </template>
+      <checkbox
+        v-model="enableTraefik"
+        :disabled="isKubernetesDisabled"
+        label="Enable Traefik"
+      />
+    </input-legend>
+  </div>
+</template>
+
+<style lang="scss" scoped>
+  .checkbox-title {
+    font-size: 1rem;
+    line-height: 1.5rem;
+    padding-bottom: 0.5rem;
+  }
+
+  .preferences-body {
+    padding: 0.75rem;
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+  }
+
+  .w-half {
+    width: 50%;
+    max-width: 20rem;
+    min-width: 20rem;
```

Comment:
It's confusing that `w-half` doesn't mean "half the width", but "20 rem".

Reply:
I think I changed my mind about half width after testing the layout for a while. I renamed this to `width-xs`.